Add function to convert size from bytes to a more readable format

lguerard opened the pull request:
Fixes #95

lguerard committed:
Add function to convert size from bytes to a more readable format

lguerard committed:
Fix Ruff linting

ehrenfeu reviewed (changes requested):
The function name `convert_bytes` isn't really telling what it's actually doing, it only reveals the bytes are converted to something unknown.

Please rename it to `human_bytes` or similar.

ehrenfeu commented:
Removed it from the current milestone as this is probably not blocking other code - feel free to re-add if you need it already now.

ehrenfeu commented:
#126

ehrenfeu commented:
I rebased this PR onto current `devel` in order to reduce the complexity of merging it (`misc.py` had many other changes already in `devel`, resulting in a lot of potential conflicts).

🚨 @lguerard please make sure to ~`git pull`~ **delete your *local* branch** and create a new one from the PR before continuing your work on this!! 🚨 

Let me know if you need help with this!

lguerard committed:
feat(misc): Rename bytes to human-readable conversion function

- Implemented `bytes_to_human_readable` function to convert byte sizes into a more readable format.
- Added unit tests for the new function to ensure accuracy across various sizes.

lguerard committed:
test: Add docstrings to byte conversion tests

* Ensure common sizes are formatted into human-readable strings.
* Verify formatting for large sizes such as terabytes.

ehrenfeu commented:
Rebased this onto current `devel` in order to resolve merge conflicts.

ehrenfeu commented on src/imcflibs/imagej/misc.py:
```diff
@@ -533,7 +531,9 @@ def write_ordereddict_to_csv(out_file, content):
+def save_image_in_format(
+    imp, format, out_dir, series, pad_number, split_channels, suffix=""
+):
     """Save an ImagePlus object in the specified format.
```
Please be more careful when crafting / assembling commits, those changes (originating from code linting and formatting tools) are completely unrelated to the PR and destroy the logical context of the code history when being committed with the message of the actual change.
